prov/verbs/src/fi_verbs.c

```diff
@@ -752,31 +861,30 @@ static int fi_ibv_getinfo(uint32_t version, const char *node, const char *servic
 			  uint64_t flags, struct fi_info *hints, struct fi_info **info)
 {
 	struct rdma_cm_id *id;
-	struct ibv_qp_init_attr qp_init_attr;
 	struct rdma_addrinfo *rai;
 	struct fi_info *fi;
+	const char *dev_name;
 	int ret;
 
+	if (!verbs_info) {
+		ret = fi_ibv_create_info(hints, &verbs_info);
+		if (ret)
+			return ret;
+	}
```

Comment:
This isn't thread safe.  At least add a comment to handle multiple threads calling fi_getinfo simultaneously.